service/src/main/java/uk/gov/hmcts/reform/fpl/handlers/cmo/AgreedCMOUploadedEventHandler.java

```diff
@@ -29,11 +31,24 @@
     @EventListener
     public void sendNotificationForAdmin(final AgreedCMOUploaded event) {
         CaseData caseData = event.getCaseData();
+        HearingBooking hearing = event.getHearing();
+
+        AbstractJudge judge = null;
+
+        if (isNotEmpty(hearing.getJudgeAndLegalAdvisor().getJudgeEmailAddress())) {
+            judge = hearing.getJudgeAndLegalAdvisor();
+        } else if (caseData.hasAllocatedJudgeEmail()) {
+            judge = caseData.getAllocatedJudge();
+        }
+
+        if (judge == null) {
+            return;
+        }
```

Comment:
we could use the getJudgeToNotify method below for consistency - or replace the getJudgeToNotify method with what you have here if you prefer it + update the sendNotificationToJudge, that works too